core/src/main/java/fr/abes/item/core/components/FichierSauvegardeSuppCsv.java

```diff
@@ -110,13 +107,25 @@ public void checkFileContent(Demande d) throws FileCheckingException, IOExceptio
     }
 
     public void writeHeader() {
-        try (FileWriter fw = new FileWriter(this.getPath().resolve(this.getFilename()).toString(), true);
-             BufferedWriter bw = new BufferedWriter(fw);
-             PrintWriter out = new PrintWriter(bw)) {
-            // ajout de la ligne
-            out.println(String.join(";", this.referenceService.constructHeaderCsv()));
-        } catch (IOException ex) {
-            throw new StorageException("Impossible d'écrire dans le fichier de sauvegarde csv");
+        if (!isHeaderExist()) {
+            try (FileWriter fw = new FileWriter(this.getPath().resolve(this.getFilename()).toString(), true);
+                 BufferedWriter bw = new BufferedWriter(fw);
+                 PrintWriter out = new PrintWriter(bw)) {
+                // ajout de la ligne
+                out.println(String.join(";", this.referenceService.constructHeaderCsv()));
+            } catch (IOException ex) {
+                throw new StorageException("Impossible d'écrire dans le fichier de sauvegarde csv");
+            }
+        }
+    }
+
+    private boolean isHeaderExist() {
+        try (FileReader fr = new FileReader(this.getPath().resolve(this.getFilename()).toString());
+             BufferedReader br = new BufferedReader(fr)) {
+            String line = br.readLine();
+            return line != null && line.startsWith("TYPE (008);PPN;");
+            } catch (IOException ex) {
+            throw new RuntimeException(ex);
```

Comment:
Il vaut mieux éviter les runtimeException et utiliser la StorageException comme dans la méthode au dessus, sinon le programme va s'arrêter brusquement

Reply:
C'est noté. Je fais une modification en ce sens et je repousse le code corrigé.